### PR TITLE
Add CrossEntropyLoss2d

### PR DESCRIPTION
**Pull request by desimone**
See #551
- Add support for `CrossEntropyLoss2d`; needs some help on the documentation front.
- `NLLLoss2d` wraps `SpatialClassNLLCriterion` and so should be able to support weights and inherit `_WeighedLoss`.
- is `_WeighedLoss` a typo? Should it be _Weigh**t**edLoss?

Per @apaszke , PR on hold till `1d`/`2d` class naming is decided.

### CODE REVIEW

**Comment by apaszke** [comment]
We've decided that we'd rather make CrossEntropyLoss universal. Can please you add that as a special case there? `nll_loss2d` should be merged into `nll_loss` too. Thanks!

**Comment by shubhamjain0594** [comment]
What is the status of this? Will love to help and close this. Let me know.

**Comment by apaszke** [comment]
@shubhamjain0594 I think it's already implemented.

**Comment by zym1010** [comment]
@apaszke  the doc of `CrossEntropyLoss` still says that it expects a 2D tensor for `input` an d 1D for `target`. Does it support multi dimensional now?

**Comment by apaszke** [comment]
It should support higher dimensions too, the docs are outdated. Note that due to legacy reasons Softmax normalizes dimension 1 (2nd), except for 3D inputs, where it uses dim 0 (1st). It will be changed soon.
